artview/components/display.py

```diff
@@ -0,0 +1,297 @@
+        gBox_layout.addWidget(self.updateButton, 0, 1, 1, 1)
+
+        self.delButton = QtGui.QPushButton("Delete Text")
+        self.delButton.clicked.connect(self.delDispText)
+        self.delButton.setToolTip('Save cfRadial data file')
+        gBox_layout.addWidget(self.delButton, 1, 0, 1, 1)
+
+        self.clrButton = QtGui.QPushButton("Clear All Text")
+        self.clrButton.clicked.connect(self.clrDispText)
+        self.clrButton.setToolTip('Remove applied filters')
+        gBox_layout.addWidget(self.clrButton, 1, 1, 1, 1)
+
+        groupBox.setLayout(gBox_layout)
+
+        return groupBox
+
+    def EntryUI(self):
+        '''Mount text entry layout.'''
+        groupBox = QtGui.QGroupBox("Text Entry and Position")
+        # groupBox.setFlat(True)
+        gBox_layout = QtGui.QGridLayout()
+
+
+        # Set up the Labels for entry
+        tex = QtGui.QLabel("Label Text")
+        xpos = QtGui.QLabel("X-Coord")
+        ypos = QtGui.QLabel("Y-Coord")
+        texc = QtGui.QLabel("Color")
+        texsz = QtGui.QLabel("Font Size")
+        texst = QtGui.QLabel("Style")
+
+        # Set up the Entry Line Entries / Add text
+        self.ent_tex = QtGui.QLineEdit('')
+        self.ent_tex.setToolTip('Label box text')
+        self.ent_xpos = QtGui.QLineEdit('')
+        self.ent_xpos.setToolTip('X position of text on display')
+        self.ent_ypos = QtGui.QLineEdit('')
+        self.ent_ypos.setToolTip('Y position of text on display')
+        self.ent_texc = QtGui.QLineEdit('')
+        self.ent_texc.setToolTip('Label text color')
+        self.ent_texsz = QtGui.QLineEdit('')
+        self.ent_texsz.setToolTip('Label text font size')
+        self.ent_texst = QtGui.QLineEdit('')
+        self.ent_texst.setToolTip('Label text font style: normal, italic, or bold')
+
+        # Add the Label and Entry fields to layout
+        gBox_layout.addWidget(tex, 0, 0, 1, 1)
+        gBox_layout.addWidget(xpos, 2, 0, 1, 1)
+        gBox_layout.addWidget(ypos, 2, 1, 1, 1)
+        gBox_layout.addWidget(texc, 4, 0, 1, 1)
+        gBox_layout.addWidget(texsz, 4, 1, 1, 1)
+        gBox_layout.addWidget(texst, 4, 2, 1, 1)
+
+        gBox_layout.addWidget(self.ent_tex, 1, 0, 1, 1)
+        gBox_layout.addWidget(self.ent_xpos, 3, 0, 1, 1)
+        gBox_layout.addWidget(self.ent_ypos, 3, 1, 1, 1)
+        gBox_layout.addWidget(self.ent_texc, 5, 0, 1, 1)
+        gBox_layout.addWidget(self.ent_texsz, 5, 1, 1, 1)
+        gBox_layout.addWidget(self.ent_texst, 5, 2, 1, 1)
+
+        groupBox.setLayout(gBox_layout)
+        return groupBox
+
+    #########################
+    #   Selection Methods   #
+    #########################
+
+    def chooseText(self, selection):
+        '''Get Display Text.'''
+##        selection = self.dispCombo.currentIndex()
+        self.choice_key = str(self.dispCombo.currentText())
+        self.choice = self.dispChoiceList[selection]
+        self._rebuild_entry()
+
+        self.dispCombo.setCurrentIndex(selection)
```

Comment:
NO, you need  `_rebuild_entry()` here, otherwise the entries are not updated by changing index.
`setCurrentIndex` is redundant, because it will call `chooseText` again. Not sure why there is not infinity loop here.